dask_jobqueue/oar.py

```diff
@@ -76,6 +99,20 @@ def __init__(
         # Add extra header directives
         header_lines.extend(["#OAR %s" % arg for arg in self.job_extra_directives])
 
+        # OAR needs to have the properties on a single line, with SQL syntaxe
+        oar_properties = [
+            header.replace("#OAR -p ", "")
+            for header in header_lines
+            if header.startswith("#OAR -p")
+        ]
+        oar_properties = list(intersperse(" AND ", oar_properties))
+        header_lines = [
+            header for header in header_lines if not header.startswith("#OAR -p")
+        ]
+        header_lines.append(
+            "#OAR -p %s" % '"' + ("".join("".join(i) for i in [oar_properties])) + '"'
+        )
+
```

Comment:
Je pense qu'il faut juste concatener ta directive "-p" et celle qui pourrait être dans `self.job_extra_directives`. Je trouve un peu tordu la façon de faire : 
- rajouter dans tous les -p dans header_lines
- puis reparcourir tous les header pour corriger

Je verrai plus la logique suivante : 
- si `memcore` ou `mem_core`, on parcourt `self.job_extra_directives` pour recherche un `-p`
- si tu trouves un `-p` tu lui rajoute `AND mem_core>=....`
- si tu ne trouve pas de `-p`, tu peux tout simplement mettre les lignes 72/74

Reply:
je parcours la liste de toutes les lignes -p car plusieurs lignes -p peuvent être présentes ici. 
un exemple peut être:
#OAR -p memcore=xxx
#OAR -p ib_rate=xxx
#OAR -p cluster=xxx

si je suis un peu ta logique, il manquerait justement le cas : "si tu trouves plusieurs lignes -p". dans ce cas, il faudra parcourir toutes ces lignes et les mettre en SQL avec des "AND". 

Après on pourra en discuter pour trouver un meilleur algo

Reply:
Je pense pas que ca soit a ton script de corriger les erreurs de syntaxe de l'utilisateur.

Pour moi, si l'utilisateur mets plusieurs `-p` dans le jobs_extra ca ne marchera pas, met ca serait à OAR de gérer l'erreur, pas à ton driver. Il ne faut pas modifier le comportement actuel d'OAR.

Reply:
Ton algo devrait juste rajouter le "-p mem_node ..." sur la dernère occurence de -p (et ne pas te préoccuper du reste).

Reply:
je comprends ton avis mtn. je comprends que c'est compliqué pour OAR de gérer plusieurs lignes -p donc j'essaye dans mon algo de les préparer au mieux pour OAR. ça évitera une mauvaise surprise pour les utilisateurs mais bon..